src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,175 @@
+            'type': 'string',
+            'help_text': _(
+                'The Installation ID from the GitHub App installation '
+                'generated by the GitHub Admin. '
+                'Example: 59980338 extracted from the installation link: '
+                'https://github.com/settings/installations/59980338 '
+                'required for creating a limited GitHub app token',
+            ),
+        },
+        {
+            'id': 'private_rsa_key',
+            'label': _('RSA Private Key'),
+            'type': 'string',
+            # `ssh_private_key` validator checks for PKCS#1 PEM-like formats
+            'format': 'ssh_private_key',
+            'secret': True,
+            'multiline': True,
+            'help_text': _(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation IDs.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'description',
+            'label': _('Description (Optional)'),
+            'type': 'string',
+            'help_text': _('To be removed after UI is updated'),
+        },
+    ],
+    'required': ['app_id', 'install_id', 'private_rsa_key'],
+}
+
+
+class EmptyKwargs(TypedDict):
+    """Schema for no keyword args."""
+
+
+class MaybeBaseURLKwarg(TypedDict, total=False):
+    """Schema for optional PyGitHub ``base_url`` keyword arg."""
+
+    base_url: str
+
+
+def extract_github_app_install_token(
+    *,
+    github_api_url: str,
+    app_id: str,
+    private_rsa_key: str,
+    install_id: str,
+    **_discarded_kwargs: Unpack[EmptyKwargs],
+) -> str:
+    """Generate a GH App Installation access token.
+
+    :param github_api_url: The GitHub instance API endpoint URL.
+    :param app_id: The GitHub App ID.
+    :param private_rsa_key: The private key associated with the GitHub
+        App.
+    :param install_id: The GitHub App Installation ID.
+    :param _discarded_kwargs: Aren't expected to be passed.
+    :returns: A GitHub access token for a GitHub App Installation.
+    :raises ValueError: If any required parameters are invalid.
+    """
+    if not app_id.isdigit():
+        raise ValueError(
+            f'Expected GitHub App ID to be an integer but got {app_id !r}',
+        )
+
+    if not install_id.isdigit():
+        raise ValueError(
+            'Expected GitHub App Installation ID to be an integer'
+            f' but got {install_id !r}',
+        )
+
+    auth = Auth.AppAuth(
+        app_id=int(app_id),
+        private_key=private_rsa_key,
+    ).get_installation_auth(installation_id=int(install_id))
+
+    extra_gh_args: MaybeBaseURLKwarg = {
+        'base_url': github_api_url,
+    } if github_api_url else {}
+    Github(  # Generate a GitHub App authentication token
+        auth=auth,
+        **extra_gh_args,
+    )
+
+    return auth.token
+
+
+github_app_lookup = CredentialPlugin(  # type: ignore[misc]  # untyped tuple
```

Comment:
#85 will allow dropping this ignore
```suggestion
github_app_lookup = CredentialPlugin(
```